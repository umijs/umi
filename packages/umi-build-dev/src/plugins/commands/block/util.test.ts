import {
  routeExists,
  genBlockName,
  depthRouterConfig,
  genRouterToTreeData,
  reduceData,
} from './util';
import routerConfig from '../../fixtures/util/routerConfig';

test('not exists', () => {
  expect(routeExists('/foo', [{ path: '/bar' }])).toEqual(false);
});

test('exists', () => {
  expect(routeExists('/foo', [{ path: '/bar' }, { path: '/foo' }])).toEqual(true);
});

test('child routes exists', () => {
  expect(routeExists('/foo', [{ routes: [{ path: '/bar' }, { path: '/foo' }] }])).toEqual(true);
});

test('pro routes exists', () => {
  expect(
    routeExists('/welcome', [
      {
        path: '/user',
        component: '../layouts/UserLayout',
        routes: [
          {
            name: 'login',
            path: '/user/login',
            component: './user/login',
          },
        ],
      },
      {
        path: '/',
        component: '../layouts/SecurityLayout',
        routes: [
          {
            path: '/',
            component: '../layouts/BasicLayout',
            authority: ['admin', 'user'],
            routes: [
              {
                path: '/',
                redirect: '/welcome',
              },
              {
                path: '/welcome',
                name: 'welcome',
                icon: 'smile',
                component: './Welcome',
              },
              {
                name: 'accountcenter',
                path: '/accountcenter',
                component: './accountcenter',
              },
              {
                component: './404',
              },
            ],
          },
          {
            component: './404',
          },
        ],
      },
      {
        component: './404',
      },
    ]),
  ).toEqual(true);
});

test('genBlockName test', () => {
  expect(genBlockName('DashboardAnalysis')).toEqual('dashboard/analysis');
});

test('gen router config', () => {
  expect(depthRouterConfig(reduceData(genRouterToTreeData(routerConfig)))).toMatchSnapshot();
});
