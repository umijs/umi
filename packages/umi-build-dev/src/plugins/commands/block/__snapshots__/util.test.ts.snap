// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gen router config 1`] = `
Array [
  Object {
    "children": Array [],
    "key": "/",
    "parentKey": null,
    "title": "/",
    "value": "/",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "key": "/data/new",
        "parentKey": "/data",
        "title": "data/new",
        "value": "/data/new",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "key": "/data/:dataId/edit",
            "parentKey": "/data/:dataId",
            "title": "data/:dataId/edit",
            "value": "/data/:dataId/edit",
          },
        ],
        "key": "/data/:dataId",
        "parentKey": "/data",
        "title": "data/:dataId",
        "value": "/data/:dataId",
      },
    ],
    "key": "/data",
    "parentKey": null,
    "title": "data",
    "value": "/data",
  },
  Object {
    "children": Array [],
    "key": "/login",
    "parentKey": null,
    "title": "login",
    "value": "/login",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "key": "/project/new",
        "parentKey": "/project",
        "title": "project/new",
        "value": "/project/new",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "key": "/project/:projectId/edit",
            "parentKey": "/project/:projectId",
            "title": "/edit",
            "value": "/project/:projectId/edit",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [],
                        "key": "/project/:projectId/safeModel/sampleset/:samplesetId/filter",
                        "parentKey": "/project/:projectId/safeModel/sampleset/:samplesetId",
                        "title": "/sampleset/:samplesetId/filter",
                        "value": "/project/:projectId/safeModel/sampleset/:samplesetId/filter",
                      },
                    ],
                    "key": "/project/:projectId/safeModel/sampleset/:samplesetId",
                    "parentKey": "/project/:projectId/safeModel/sampleset",
                    "title": "/sampleset/:samplesetId",
                    "value": "/project/:projectId/safeModel/sampleset/:samplesetId",
                  },
                ],
                "key": "/project/:projectId/safeModel/sampleset",
                "parentKey": "/project/:projectId/safeModel",
                "title": "/sampleset",
                "value": "/project/:projectId/safeModel/sampleset",
              },
              Object {
                "children": Array [],
                "key": "/project/:projectId/safeModel/virtualFuse",
                "parentKey": "/project/:projectId/safeModel",
                "title": "/virtualFuse",
                "value": "/project/:projectId/safeModel/virtualFuse",
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "key": "/project/:projectId/safeModel/securityModel/new",
                    "parentKey": "/project/:projectId/safeModel/securityModel",
                    "title": "/securityModel/new",
                    "value": "/project/:projectId/safeModel/securityModel/new",
                  },
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [],
                        "key": "/project/:projectId/safeModel/securityModel/:modelId/edit",
                        "parentKey": "/project/:projectId/safeModel/securityModel/:modelId",
                        "title": "/securityModel/:modelId/edit",
                        "value": "/project/:projectId/safeModel/securityModel/:modelId/edit",
                      },
                    ],
                    "key": "/project/:projectId/safeModel/securityModel/:modelId",
                    "parentKey": "/project/:projectId/safeModel/securityModel",
                    "title": "/securityModel/:modelId",
                    "value": "/project/:projectId/safeModel/securityModel/:modelId",
                  },
                ],
                "key": "/project/:projectId/safeModel/securityModel",
                "parentKey": "/project/:projectId/safeModel",
                "title": "/securityModel",
                "value": "/project/:projectId/safeModel/securityModel",
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "key": "/project/:projectId/safeModel/virtualDataSet/new",
                    "parentKey": "/project/:projectId/safeModel/virtualDataSet",
                    "title": "/virtualDataSet/new",
                    "value": "/project/:projectId/safeModel/virtualDataSet/new",
                  },
                  Object {
                    "children": Array [],
                    "key": "/project/:projectId/safeModel/virtualDataSet/:sampleId",
                    "parentKey": "/project/:projectId/safeModel/virtualDataSet",
                    "title": "/virtualDataSet/:sampleId",
                    "value": "/project/:projectId/safeModel/virtualDataSet/:sampleId",
                  },
                ],
                "key": "/project/:projectId/safeModel/virtualDataSet",
                "parentKey": "/project/:projectId/safeModel",
                "title": "/virtualDataSet",
                "value": "/project/:projectId/safeModel/virtualDataSet",
              },
            ],
            "key": "/project/:projectId/safeModel",
            "parentKey": "/project/:projectId",
            "title": "/safeModel",
            "value": "/project/:projectId/safeModel",
          },
        ],
        "key": "/project/:projectId",
        "parentKey": "/project",
        "title": "project/:projectId",
        "value": "/project/:projectId",
      },
    ],
    "key": "/project",
    "parentKey": null,
    "title": "project",
    "value": "/project",
  },
]
`;
